logger.js

```diff
@@ -10,135 +10,45 @@
  * LICENSE file.
  *
  * Removal or modification of this copyright notice is prohibited.
+ *
  */
+const fs = require('fs');
+const path = require('path');
+const bunyan = require('bunyan');
 
-'use strict';
-
-var fs = require('fs');
-var util = require('util');
-var child_process = require('child_process');
-var path = require('path');
-var strftime = require('strftime').utc();
-
-require('colors');
-
-module.exports = function(config) {
-	config = config || {};
-	var exports = {};
-
-	config.levels = config.levels || {
-		none: 99,
-		trace: 0,
-		debug: 1,
-		log: 2,
-		info: 3,
-		warn: 4,
-		error: 5,
-		fatal: 6,
-	};
-
-	config.level_abbr = config.level_abbr || {
-		trace: 'trc',
-		debug: 'dbg',
-		log: 'log',
-		info: 'inf',
-		warn: 'WRN',
-		error: 'ERR',
-		fatal: 'FTL',
-	};
-
-	config.filename = `${process.cwd()}/${config.filename || 'logs.log'}`;
-
-	config.errorLevel = config.errorLevel || 'log';
-
-	child_process.execSync(`mkdir -p ${path.dirname(config.filename)}`);
-	var log_file = fs.createWriteStream(config.filename, { flags: 'a' });
-
-	exports.setLevel = function(errorLevel) {
-		config.errorLevel = errorLevel;
-	};
+const defaultLoggerConfig = {
+	filename: 'logs/lisk.log',
+	level: 'none',
+};
 
-	function snipFragileData(data) {
-		for (var key in data) {
-			if (key.search(/passphrase|password/i) > -1) {
-				data[key] = 'XXXXXXXXXX';
-			}
-		}
-		return data;
+const createDirIfNotExists = filename => {
+	if (!filename || filename === '') {
```

Comment:
if (!filename) handles all check `null, undefined and empty string` so we can remove `|| filename === ''`